### PR TITLE
Don't assume color=always when explicitally specified

### LINKED ISSUE #31546: --color=always still checks for a terminal before printing colors
rustc --color=always foo.rs 2>&1 | cat

This produces colored output on 1.6.0, but not 1.7.0 or 1.8.0. Related to rust-lang/rust-playpen#174, cc @nrc.

### PR DESCRIPTION
**Pull request by Stebalien**
Fixes #31546

### CODE REVIEW

**Comment by Stebalien** [comment]
Note: I haven't actually tested this change because rust takes an eternity to build on my poor old laptop. However, I don't expect it to fail (famous last words...).

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -934,17 +934,17 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
             Some("human")   => ErrorOutputType::HumanReadable(color),
             Some("json") => ErrorOutputType::Json,
 
-            None => ErrorOutputType::default(),
+            None => ErrorOutputType::HumanReadable(color),
 
             Some(arg) => {
-                early_error(ErrorOutputType::default(), &format!("argument for --error-format must \
+                early_error(ErrorOutputType::HumanReadable(color), &format!("argument for --error-format must \
```
Comment (nrc) [inline src/librustc/session/config.rs]: Overlong line

Comment (Stebalien) [inline src/librustc/session/config.rs]: Fixed.

File: src/librustc/session/config.rs [hunk 2]
```diff
@@ -934,17 +934,18 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
...
 
             Some(arg) => {
-                early_error(ErrorOutputType::default(), &format!("argument for --error-format must \
-                                                                  be human or json (instead was \
-                                                                  `{}`)",
-                                                                 arg))
+                early_error(ErrorOutputType::HumanReadable(color),
+                            &format!("argument for --error-format must \
+                                      be human or json (instead was \
+                                      `{}`)",
```
Comment (nrc) [inline src/librustc/session/config.rs]: nit: this can squeeze on the previous line.

**Comment by nrc** [comment]
r+ with the style nit addressed

**Comment by alexcrichton** [comment]
Thanks for the quick fix @Stebalien!

Nominating for backport as well

**Comment by durka** [comment]
Can we test this somehow to prevent future regressions? Maybe with a run-make test that greps for \x1b or something?

**Comment by pnkfelix** [comment]
@durka I agree, it seems like testing this via `run-make` would be good.  Adding E-needstest flag...

**Comment by nikomatsakis** [comment]
Accepting for beta since seems awfully minor.

**Comment by nikomatsakis** [comment]
cc @rust-lang/compiler

**Comment by arielb1** [comment]
+1

**Pull request merged**